lib/services/notifications_cache.go

```diff
@@ -210,21 +210,25 @@ func (c *UserNotificationCache) StreamUserNotifications(ctx context.Context, use
 		startKey = fmt.Sprintf("%s/%s", username, startKey)
 	}
 
+	const limit = 50
```

Comment:
I know this was hardcoded to 50 before, but there is a `PageSize` int in `ListNotificationsRequest`. I wonder why that isn't taken into account anywhere.

Nothing to do in this PR of course, just talking out loud